From 72d99dcd1d88393552a8c4902be50c34ec3a6689 Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Fri, 7 Dec 2018 07:22:43 +0100
Subject: [PATCH] Merge pull request #310 in SW_UGW/linux from
 feature/UGW_SW-33065-linux-security-incident-fragmentsmack-update-and-push-kenel-patch_
 to master

* commit '838ba43e5b28e2b12cf7638650dbc95323bb6454':
  fragmentsmack kernel patch :
---
 include/net/ipv6.h     | 10 ++++++++--
 net/ipv4/ip_fragment.c |  9 +++++++--
 2 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/include/net/ipv6.h b/include/net/ipv6.h
index e64210c98c2b..2621cdf26294 100644
--- a/include/net/ipv6.h
+++ b/include/net/ipv6.h
@@ -344,8 +344,14 @@ static inline int ip6_frag_mem(struct net *net)
 }
 #endif
 
-#define IPV6_FRAG_HIGH_THRESH	(4 * 1024*1024)	/* 4194304 */
-#define IPV6_FRAG_LOW_THRESH	(3 * 1024*1024)	/* 3145728 */
+/* Fragmentsmack CVE-2018-5391 */
+#define IPV6_FRAG_HIGH_THRESH	(256 * 1024)
+#define IPV6_FRAG_LOW_THRESH	(192 * 1024)
+/*
+ * OLD default values
+ * #define IPV6_FRAG_HIGH_THRESH (4 * 1024*1024) 4194304
+ * #define IPV6_FRAG_LOW_THRESH  (3 * 1024*1024) 3145728
+ */
 #define IPV6_FRAG_TIMEOUT	(60 * HZ)	/* 60 seconds */
 
 int __ipv6_addr_type(const struct in6_addr *addr);
diff --git a/net/ipv4/ip_fragment.c b/net/ipv4/ip_fragment.c
index 4bf3b8af0257..4f4a152cbde3 100644
--- a/net/ipv4/ip_fragment.c
+++ b/net/ipv4/ip_fragment.c
@@ -858,8 +858,13 @@ static int __net_init ipv4_frags_init_net(struct net *net)
 	 * we will prune down to 3MB, making room for approx 8 big 64K
 	 * fragments 8x128k.
 	 */
-	net->ipv4.frags.high_thresh = 4 * 1024 * 1024;
-	net->ipv4.frags.low_thresh  = 3 * 1024 * 1024;
+        net->ipv4.frags.high_thresh = 256 * 1024; /* Fragmentsmack CVE-2018-5391 */
+        net->ipv4.frags.low_thresh  = 192 * 1024;
+	/*
+         * old default values
+         * net->ipv4.frags.high_thresh =  4 * 1024 * 1024 ;
+	 * net->ipv4.frags.low_thresh  =  3 * 1024 * 1024;
+         */
 	/*
 	 * Important NOTE! Fragment queue must be destroyed before MSL expires.
 	 * RFC791 is wrong proposing to prolongate timer each fragment arrival
